src/validation/date.validation.ts

```diff
@@ -222,7 +222,10 @@ const validateUnderageRule = (dateOfBirth: Date, dateValidationType: DateValidat
  * @returns A ValidationError object if one occurred, else undefined
  */
 const validateUnderageAtAppointmentRule = (dateOfBirth: Date, dateOfAppointment: Date,dateValidationType: DateValidationType): ValidationError | undefined => {
-  if(getAge(dateOfBirth) - getAge(dateOfAppointment) < 16){
+  const doaTime = dateOfAppointment.getTime();
+  const dobTime = dateOfBirth.getTime();
+
+  if( ((doaTime - dobTime) / (1000 * 3600 * 24 * 365.25)) < 16){
```

Comment:
Maybe add a test to validate the leap year 365.25 consideration?

Reply:
I can do, but in this case it shouldn't matter, as we are talking about 16 years,  it means either both are leap years or both are not leap years.  if its not that then we are talking a whole year out at least so a leap year day wouldn't be relevant.

Reply:
tests added.